geopandas/tests/test_op_output_types.py

```diff
@@ -0,0 +1,209 @@
+        geo_name = df.geometry.name
+        self._assert_object(df.iloc[:, 0:2], pd.DataFrame)
+        self._assert_object(df.iloc[:, 2:4], GeoDataFrame, geo_name)
+        self._assert_object(df.iloc[:, [2]], GeoDataFrame, geo_name)
+        self._assert_object(df.iloc[:, [3, 0]], GeoDataFrame, None)
+        self._assert_object(df.iloc[:, [3]], GeoDataFrame, None)
+        # Ideally this would mirror __getitem__ and below would be true
+        # self._assert_object(df.iloc[:, [3, 0]], pd.DataFrame)
+        # self._assert_object(df.iloc[:, [3]], pd.DataFrame)
+        self._assert_object(df.iloc[:, [0]], pd.DataFrame)
+        # Series
+        self._assert_object(df.iloc[:, 2], GeoSeries, geo_name)
+        self._assert_object(df.iloc[:, 3], GeoSeries, "geometry2", crs=crs_osgb)
+        self._assert_object(df.iloc[:, 0], pd.Series)
+
+    def test_squeeze(self, df):
+        geo_name = df.geometry.name
+        self._assert_object(df[[geo_name]].squeeze(), GeoSeries, geo_name)
+
+        # Not ideal behaviour, but this is consistent with __getitem__
+        self._assert_object(df[["geometry2"]].squeeze(), pd.Series)
+
+    def test_to_frame(self, df):
+        geo_name = df.geometry.name
+        # TODO this reflects current behaviour, but we should fix
+        #  GeoSeries._constructor_expanddim so this doesn't happen
+        res1 = df[geo_name].to_frame()
+        if geo_name == "geometry":  # -> this should be doable for any geo_name
+            self._assert_object(res1, GeoDataFrame, geo_name)
+        assert res1._geometry_column_name == "geometry"  # -> should be geo_name
+
+        res2 = df["geometry2"].to_frame()
+
+        assert type(res2) is GeoDataFrame
+        assert res2._geometry_column_name == "geometry"  # -> should be geometry2
+        assert res2.crs is None  # -> should be crs_osgb
+        # also res2.geometry should not crash because geometry isn't set
+        self._assert_object(df["value1"].to_frame(), pd.DataFrame)
+
+    def test_reindex(self, df):
+        geo_name = df.geometry.name
+
+        test_func = self._assert_object  # easier to read without black line wraps
+        test_func(df.reindex(columns=["value1", "value2"]), pd.DataFrame)
+        test_func(df.reindex(columns=[geo_name, "geometry2"]), GeoDataFrame, geo_name)
+        test_func(df.reindex(columns=[geo_name]), GeoDataFrame, geo_name)
+        test_func(df.reindex(columns=["geometry2", "value1"]), GeoDataFrame, None)
+        test_func(df.reindex(columns=["geometry2"]), GeoDataFrame, None)
+        # Ideally this would mirror __getitem__ and below would be true
+        # test_func(df.reindex(columns=["geometry2", "value1"]), pd.DataFrame)
+        # test_func(df.reindex(columns=["geometry2"]), pd.DataFrame)
+        test_func(df.reindex(columns=["value1"]), pd.DataFrame)
+
+    def test_drop(self, df):
+        geo_name = df.geometry.name
+        test_func = self._assert_object  # easier to read without black line wraps

```

Comment:
I will push a small change to make the tests functions instead of methods in a class, that will also help with black line length, without needing to use this `test_func`